mypy/checker.py

```diff
@@ -1076,7 +1076,8 @@ def check_assignment(self, lvalue: Node, rvalue: Node, infer_lvalue_type: bool =
                 elif (is_literal_none(rvalue) and
                         isinstance(lvalue, NameExpr) and
                         isinstance(lvalue.node, Var) and
-                        lvalue.node.is_initialized_in_class):
+                        lvalue.node.is_initialized_in_class and
+                        not experiments.STRICT_OPTIONAL):
```

Comment:
The idea here is to prohibit things like ``x: int = None``. Probably you are right that this could be done by just removing the check. I will try this and if everything will work, I just remove this check in the next commit.

Concerning the AST replacement in the parser, see my last comment below.